Implement `--pytest` option to `dbt-build`

andrewmogan opened the pull request:
Addresses #295. Running `dbt-build --pytest` will scan the `$DBT_AREA_ROOT/sourcecode` directory for pytest modules stored in an `integtest` directory. Packages with tests stored elsewhere are not yet accounted for, but could be added in the future. `dbt-build --pytest <package>` is also supported. 

The pytest output is stored in both a log file and a junit xml file. The junit xml file will be parsed in an upcoming workflow for running pytests. There is also a summary file that shows which tests had one or more failures.

andrewmogan committed:
Implement dummy --pytest option for dbt-build

andrewmogan committed:
Make pytest args consistent and add more logging

andrewmogan committed:
Loop over individual packages for pytest logging purposes

andrewmogan committed:
Remove ignore_args and improve logging

jcfreeman2 commented:
This looks great, just a couple of minor suggestions:
* If I run `dbt-build --clean --pytest --unittest` with a `listrev` repo, I don't see the usual pytest output that I'd see if I were to run `pytest -s $LISTREV_SHARE/integtest/listrev_test.py`, and it doesn't appear in the log either:
```
============================= test session starts ==============================
platform linux -- Python 3.10.10, pytest-8.3.3, pluggy-1.5.0
rootdir: /home/nfs/jcfree/NFD_DEV_250305_A9/sourcecode
plugins: anyio-4.6.2.post1, integrationtest-3.1.0
collected 14 items

listrev/integtest/listrev_test.py ..............                         [100%]

- generated xml file: /home/nfs/jcfree/NFD_DEV_250305_A9/log/pytest_tests_Wed_Mar_5_09_44_02_AM_CST_2025/listrev_listrev_test_results.xml -
======================== 14 passed in 276.92s (0:04:36) ========================
```
It might be nice to provide this output, since especially for a test that runs for several minutes it's hard to tell if there's been a hang or not
* When I tried running, a `__pycache__` subdirectory appeared in `sourcecode/listrev/integtest/__pycache__`; it would be preferable not to have this
* It also seems like a `.pytest_cache` directory appears in `sourcecode`; could we move this a subdirectory up? Otherwise when I `dbt-build` I see stuff like `CMake Warning at /home/nfs/jcfree/daq-buildtools/cmake/DAQTopprojHelpers.cmake:30 (_message):
  Skipping directory .pytest_cache: No CMakeLists.txt found.`

andrewmogan committed:
Improve verbosity and remove __pycache__ dirs in pytests

andrewmogan committed:
Address PR comments

- Use pathlib.Path for easier path handling
- (Re)move pytest cache directories
- Clean up some output, remove comments, etc.

jcfreeman2 reviewed (approved):
I've confirmed that all my comments were addressed; approved.
